[Pawirunsiri Thanakorn] iP

modembcc opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Panda is
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

***

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
    new Panda(filePath).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

modembcc committed:
Level 0

modembcc committed:
Level-1

modembcc committed:
Level-2

modembcc committed:
Level-3

modembcc committed:
Level-4

modembcc committed:
A-TextUiTesting

modembcc committed:
Level-5

modembcc committed:
Level-6

modembcc committed:
Level-7

modembcc committed:
Merge branch 'branch-level-7'

modembcc committed:
Level-8

modembcc committed:
Merge branch 'branch-Level-8'

modembcc committed:
A-MoreOOP

modembcc committed:
Merge branch 'branch-A-MoreOOP'

modembcc committed:
A-Packages

modembcc committed:
A-Packages

modembcc committed:
Merge branch 'branch-A-Packages'

modembcc committed:
Merge branch 'add-gradle-support'

modembcc committed:
A-Gradle

modembcc committed:
Add test and Override command equals

modembcc committed:
Merge branch 'branch-A-JUnit'

modembcc committed:
Create Jar

modembcc committed:
Add JavaDoc

modembcc committed:
Tweak the code to comply with CS2103T coding standard

modembcc committed:
Merge branch 'branch-A-CodingStandard'

modembcc committed:
Add find method to find tasks

modembcc committed:
Merge branch 'branch-Level-9'

modembcc committed:
Merge branch 'branch-A-JavaDoc'

modembcc committed:
Merge branch 'branch-A-CodingStandard'

modembcc committed:
Merge branch 'branch-Level-9'

modembcc committed:
Merge branch 'master' of https://github.com/modembcc/ip

modembcc committed:
Resolve merge conflict

Yskie commented on src/main/java/panda/Panda.java:
```diff
@@ -0,0 +1,61 @@
+
+public class Panda {
+
+    private TaskList tlist;
```
Would it be better to have a more descriptive name such as taskList or simply tasks?

Yskie commented on src/main/java/panda/component/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @return the parsed Command.
+     * @throws PandaException if the user input does not correspond to a known command.
+     */
+    public static Command parse(String userInput) throws PandaException {
```
Would it be better if the method name is parseUserInputToCommand?

Yskie commented on src/main/java/panda/command/AlterMarkCommand.java:
```diff
@@ -0,0 +1,85 @@
+import panda.exception.OutOfBoundsException;
+import panda.component.*;
+public class AlterMarkCommand extends Command {
+    private int idx;
```
Would it be better for "idx" to be "taskIndex" for clarity?

Yskie commented on src/main/java/panda/command/DeleteCommand.java:
```diff
@@ -0,0 +1,76 @@
+import panda.exception.OutOfBoundsException;
+import panda.component.*;
+public class DeleteCommand extends Command {
+    private int idx;
```
Would it be better for "idx" to be "taskIndex" for clarity? The same applies to the rest of the similar code.

Yskie commented on src/main/java/panda/command/FindCommand.java:
```diff
@@ -0,0 +1,70 @@
+     * @param cacheFile the cache file to save changes to.
+     */
+    public void execute(TaskList tlist, Ui ui, Storage cacheFile) {
+        TaskList ftlist = tlist.find(fString);
```
Would it be better to give a more meaningful name to ftList, so that others could understand what kind of TaskList instance ftList is.

Yskie commented on src/main/java/panda/component/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     * Initializes the Scanner for reading user inputs.
+     */
+    public Ui() {
+        myScanner  = new Scanner(System.in);
```
Maybe you could consider changing "myScanner" to simply "scanner"?

Yskie commented on src/main/java/panda/task/Task.java:
```diff
@@ -0,0 +1,88 @@
+     * 
+     * @return the string representation of the task.
+     */
+    public String saveString() {
```
Would it be better to name this method as toSaveString() as you are not actually saving the string in this method?

Yskie commented on src/main/java/panda/component/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+     * 
+     * @param task the task to insert.
+     */
+    public void insert(Task task) {
```
Would insertTask or addTask be a more descriptive name for this method?

Yskie reviewed (commented):
Generally looks good, would be much better if the variable and method names are more descriptive.

theman-oj10 commented on src/main/java/panda/command/AlterMarkCommand.java:
```diff
@@ -0,0 +1,85 @@
+        if(isMarked) {
+            tlist.mark(idx);
+        }
+        else {
```
Is the first execute method still needed since the updated code is using the new Ui and Storage classes?

theman-oj10 commented on src/main/java/panda/command/FindCommand.java:
```diff
@@ -0,0 +1,70 @@
+     */
+    @Override
+    public boolean equals(Object o) {
+
```
Minor nitpick but I think removing the empty lines in this method can help with better formatting.

theman-oj10 commented on src/main/java/panda/component/Parser.java:
```diff
@@ -0,0 +1,106 @@
+            }
+            return new NewTaskCommand(new Todo(splitted[1].trim()));
+        }
+        if(userInput.split(" ")[0].equals("deadline")) {
```
Perhaps using switch statements can make the code more concise?

theman-oj10 commented on src/main/java/panda/component/Storage.java:
```diff
@@ -0,0 +1,95 @@
+                if(parts[0].equals("T")) {
+                    clist.add(new NewTaskCommand(new Todo(parts[2])));
+                } else  {
+                    if(parts[0].equals("D")) {
```
Perhaps use else if statements or switch statements to make the code more concise?

theman-oj10 commented on src/main/java/panda/component/Ui.java:
```diff
@@ -0,0 +1,89 @@
+        return userInput.trim();
+    }
+
+    // private void reply(String replyString) {
```
Perhaps, these unused methods can be removed?

theman-oj10 commented on src/main/java/panda/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+     * @return the string representation of the Deadline.
+     */
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + dateString() + ")";
```
Perhaps String.format() can be used for cleaner code? Just a minor nitpick.

theman-oj10 reviewed (commented):
LGTM, just a few nits with the formatting. Overall, Good Job!

modembcc committed:
Add JavaFX to Panda

modembcc committed:
Merge branch 'branch-Level-10'

modembcc committed:
Add java assertions to commands

As the program is designed to handle most of the misinputs, the only place java assertions make sense is in the commands.

Executing the commands while passing an uninitialized taskList is a sign that the program is already in a broken state. Thus assertions is used instead of catching for errors.

modembcc committed:
Improve code quality

The code has been changed in various ways to improve readability.

The conditional statements in Storage.java has been refactored to match the coding standards. The long Parser.parse method has been refactored to be shorter while keeping SLAP.  The package and import statements have been made consistent throughout the codebase. Unused code has been removed.

modembcc committed:
Merge pull request #1 from modembcc/branch-A-Assertions

Add java assertions to commands

modembcc committed:
Merge remote-tracking branch 'origin' into branch-A-CodeQuality

modembcc committed:
Add some missed assertions

modembcc committed:
Merge pull request #2 from modembcc/branch-A-CodeQuality

Improve code quality

modembcc committed:
Fix checkstyle issues

Change remaining unchanged braces to egyptian style braces.

Change javadoc comments to correct format.

modembcc committed:
Add tagging system

modembcc committed:
Merge branch 'branch-BCD-Extension'

modembcc committed:
Fix jar file

Previous release was unusable because of storage file path mismatch and incompatible java versions.

modembcc committed:
Improve Gui using css

modembcc committed:
Fix parser bug for invalid command case

modembcc committed:
Fix delete task

modembcc committed:
Improve UI

The pictures and titles were changed to match with the name Panda.

Add a welcome message and an exit message. Also add command list.

modembcc committed:
Remove Duke from code

modembcc committed:
Add user guide

modembcc committed:
Fix Ui Picture not showing
